Yosemite/Yosemite/Tools/Plugins/PluginsService.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+import WooFoundation
+import protocol Storage.StorageManagerType
+
+/// A service for system plugins.
+public protocol PluginsServiceProtocol {
+    /// Waits for a specific plugin to be available in storage.
+    func waitForPluginInStorage(siteID: Int64, pluginName: String, isActive: Bool) async -> SystemPlugin
+}
+
+public class PluginsService: PluginsServiceProtocol {
+    private let storageManager: StorageManagerType
+
+    public init(storageManager: StorageManagerType) {
+        self.storageManager = storageManager
+    }
+
+    @MainActor
+    public func waitForPluginInStorage(siteID: Int64, pluginName: String, isActive: Bool) async -> SystemPlugin {
```

Comment:
@jaclync Just noticed this while debugging, sorry for the late comment...

Checking plugin names has bitten us in the past, e.g. when WooCommerce Payments became WooPayments. 

This service might be safer if we check the `plugin` property instead, which has the slug. We'd have to double check that's the right way, but I think it is...